climbing_stairs/climbing_stairs.py

```diff
@@ -2,8 +2,15 @@
 
 import sys
 
-def climbing_stairs(n, cache=None):
-  pass
+def climbing_stairs(n):
+    if n <= 1:
+        return 1
+    elif n == 2:
+        return 2
+    elif n == 3:
+        return 4
+    else:
+        return climbing_stairs(n-1) + climbing_stairs(n-2) + climbing_stairs(n-3)
```

Comment:
awesome job man! you can tell by just looking at the code that this will work, it's super clean! I ran it and it obviously gives out the right answers. 
the only thing you're missing is using a cache in order to make this much quicker. similar to the fibonacci problem, this will take forever on larger numbers. but you've got the hard part down. great!